src/Swarm/Game/Entity.hs

```diff
@@ -448,9 +450,49 @@ entityInventory = hashedLens _entityInventory (\e x -> e {_entityInventory = x})
 -- Inventory
 ------------------------------------------------------------
 
--- | A convenient synonym to remind us when an 'Int' is supposed to
---   represent /how many/ of something we have.
-type Count = Int
+-- | A type that represent /how many/ of something we have.
+--
+-- In some challenges it is useful to have _an infinite_ amount
+-- of supplies in robots inventory as that is more visible
+-- than hidden default devices.
+data Count = Count Natural | Infinity
+  deriving (Eq, Ord, Show, Read, Generic, Data)
+
+instance ToJSON Count where
+  toJSON (Count a) = toJSON a
+  toJSON Infinity = Null
+
+instance FromJSON Count where
+  parseJSON = \case
+    Number s ->
+      if s - fromInteger (truncate s) == 0
+        then pure $ Count (truncate s)
+        else fail "Count is not a whole number!"
+    Null -> pure Infinity
+    e ->
+      fail $
+        "Expected number or null for count, but got '" <> show e <> "'!"
+
+instance FromJSONE e Count
+
+instance Num Count where
+  Count a + Count b = Count $ a + b
+  _ + _ = Infinity
+  Count a * Count b = Count $ a * b
+  _ * _ = Infinity
+  abs = id
+  signum (Count 0) = 0
+  signum _ = 1
+  fromInteger = Count . fromInteger
+  Count a - Count b = Count $ a - b
```

Comment:
On `3 - 5` it raises an underflow error.

Reply:
Eek!  Yeah, I definitely don't want Natural subtraction in there if it can crash the game.  We could make our own `Natural` type which uses saturating subtraction (i.e.  3 - 5 = 0).

Reply:
Solved by using negative infinity. :heavy_check_mark: